src/olympia/activity/views.py

```diff
@@ -19,12 +19,20 @@ class VersionReviewNotesViewSet(AddonChildMixin, RetrieveModelMixin,
     ]
     serializer_class = ActivityLogSerializer
     queryset = ActivityLog.objects.all()
-    filter = amo.LOG_REVIEW_QUEUE_DEVELOPER
 
     def get_queryset(self):
         addon = self.get_addon_object()
         version_object = get_object_or_404(
             Version.unfiltered.filter(addon=addon),
             pk=self.kwargs['version_pk'])
         alog = ActivityLog.objects.for_version(version_object)
-        return alog.filter(action__in=self.filter)
+        return alog.filter(action__in=amo.LOG_REVIEW_QUEUE_DEVELOPER)
+
+    def get_addon_object(self):
+        return super(VersionReviewNotesViewSet, self).get_addon_object(
+            permission_classes=self.permission_classes)
+
+    def check_object_permissions(self, request, obj):
+        """Check object permissions against the Addon, not the ActivityLog."""
+        # Just loading the add-on object should trigger permission checks.
+        self.get_addon_object()
```

Comment:
I don't like the word "should" here, let's rather mention `AddonChildMixin` for more details maybe? Or be more explicit and say "Loading the add-on triggers proper permission checks."